cloud-consumption/src/test/java/com/sequenceiq/consumption/job/storage/StorageConsumptionJobServiceTest.java

```diff
@@ -0,0 +1,201 @@
+        doReturn(null).when(scheduler).getJobDetail(argThat((JobKey jobkey) -> "1".equals(jobkey.getName())));
+
+        underTest.unschedule(consumptionToUnSchedule);
+
+        verify(scheduler, never()).deleteJob(any());
+    }
+
+    @Test
+    public void testUnscheduleWithAggregationJobRunningForConsumptionNoReschedule() throws SchedulerException {
+        Consumption consumptionToUnSchedule = consumption(1L);
+        when(consumptionService.isAggregationRequired(consumptionToUnSchedule)).thenReturn(true);
+        doReturn(mock(JobDetail.class)).when(scheduler).getJobDetail(argThat((JobKey jobkey) -> "1".equals(jobkey.getName())));
+        when(consumptionService.findAllStorageConsumptionForEnvCrnAndBucketName(
+                consumptionToUnSchedule.getEnvironmentCrn(), consumptionToUnSchedule.getStorageLocation()))
+                .thenReturn(List.of(consumptionToUnSchedule));
+
+        underTest.unschedule(consumptionToUnSchedule);
+
+        verify(scheduler, times(1)).deleteJob(jobKeyCaptor.capture());
+        assertEquals("1", jobKeyCaptor.getValue().getName());
+        verify(scheduler, never()).scheduleJob(any(), any());
+    }
+
+    @Test
+    public void testUnscheduleWithAggregationJobRunningForConsumptionWithReschedule() throws SchedulerException {
+        Consumption consumptionToUnSchedule = consumption(1L);
+        Consumption consumptionOther = consumption(2L);
+        when(consumptionService.isAggregationRequired(consumptionToUnSchedule)).thenReturn(true);
+        doReturn(mock(JobDetail.class)).when(scheduler).getJobDetail(argThat((JobKey jobkey) -> "1".equals(jobkey.getName())));
+        when(consumptionService.findAllStorageConsumptionForEnvCrnAndBucketName(
+                consumptionToUnSchedule.getEnvironmentCrn(), consumptionToUnSchedule.getStorageLocation()))
+                .thenReturn(List.of(consumptionToUnSchedule, consumptionOther));
+        mockSchedulingById(2L);
+
+        underTest.unschedule(consumptionToUnSchedule);
+
+        verify(scheduler, times(1)).deleteJob(jobKeyCaptor.capture());
+        assertEquals("1", jobKeyCaptor.getValue().getName());
+        verify(scheduler, times(1)).scheduleJob(jobDetailCaptor.capture(), any());
+        assertEquals("2", jobDetailCaptor.getValue().getKey().getName());    }
+
+    private void mockJobResource(Long id) {
+        JobResource jobResource = mock(JobResource.class);
+        when(jobResource.getLocalId()).thenAnswer(invocation -> id.toString());
+        when(jobResource.getRemoteResourceId()).thenReturn("crn");
+        when(consumptionRepository.getJobResource(id)).thenReturn(Optional.of(jobResource));
+    }
+
+    private void mockSchedulingById(Long id) throws SchedulerException {
+        when(storageConsumptionConfig.isStorageConsumptionEnabled()).thenReturn(true);
+        mockJobResource(id);
+        doReturn(null).when(scheduler).getJobDetail(argThat((JobKey jobkey) -> id.toString().equals(jobkey.getName())));
```

Comment:
I think `doReturn(null)` is the default behavior when the return type is not a primitive type, array or collection.